DEV-GUIDELINES.md

```diff
@@ -0,0 +1,173 @@
+Development Guidelines
+------------
+
+This document provides a list of rules and best practices followed by project Helidon.
+Please follow these rules when contributing to the project, when refactoring existing code and when
+reviewing changes done by others.
+
+**Exceptions to these rules should be documented clearly.**
+
+# Package and module structure
+1. We use flat package structure
+    1. Each module (maven and jigsaw) has a single implementation package
```

Comment:
Is it true that each Maven module is a Jigsaw module? If so, maybe we should say that explicitly.

Reply:
Added to guidelines